windows/drivers/uwb/simulator/UwbSimulatorDevice.cxx

```diff
@@ -318,3 +290,16 @@ UwbSimulatorDevice::GetSessionCount()
     std::shared_lock sessionsReadLock{ m_sessionsGate };
     return std::size(m_sessions);
 }
+
+void
+UwbSimulatorDevice::PushUwbNotification(UwbNotificationData uwbNotificationData)
+{
+    // TODO: log
+    // TODO: lock

```

Comment:
I added a `std::mutex` for this since the first iteration and removed the TODO. 

But to your point: the locking is necessary here, irrespective of what happens in `UwbSimulatorIoEventQueue::PushNotification()` because we're reading `m_deviceFiles` which will be modified any time a handle to the driver is opened, so access to it needs to be synchronized.

However, looking at this again, we really shouldn't continue to hold the lock while executing `PushNotification` for each device file, since we'll already have obtained a `shared_ptr` at that point. I'll make an improvement here.

Reply:
Sorry, my comment is unintelligible. I was actually referring to changes I haven't sent yet which change the pointer type from raw to weak_ptr. I'll make this change on that next PR.